libraries/chain/market_evaluator.cpp

```diff
@@ -167,6 +169,10 @@ void_result call_order_update_evaluator::do_evaluate(const call_order_update_ope
    FC_ASSERT( _debt_asset->is_market_issued(), "Unable to cover ${sym} as it is not a collateralized asset.",
               ("sym", _debt_asset->symbol) );
 
+   FC_ASSERT( next_maintenance_time <= HARDFORK_CORE_1465_TIME 
+         || _debt_asset->dynamic_data(d).current_supply + o.delta_debt.amount <= _debt_asset->options.max_supply,
```

Comment:
Please save a pointer to the dynamic_data object in the class to avoid querying for it again in `do_apply`, although the query performance has been improved due to #1462.

By the way, perhaps better move the `current_supply >= 0` check to here from `do_apply`.

Reply:
@abitmore done

Reply:
Then please use the pointer here.

Reply:
Fixed.